src/builtins/echo.rs

```diff
@@ -0,0 +1,135 @@
+use std::io::{self, Write, BufWriter};
+
+const MAN_PAGE: &'static str = /* @MANSTART{echo} */ r#"
+NAME
+    echo - display a line of text
+SYNOPSIS
+    echo [ -h | --help ] [-e] [-n] [-s] [STRING]...
+DESCRIPTION
+    Print the STRING(s) to standard output.
+OPTIONS
+    -e
+        enable the interpretation of backslash escapes
+    -n
+        do not output the trailing newline
+    -s
+        do not separate arguments with spaces
+    Escape Sequences
+        When the -e argument is used, the following sequences will be interpreted:
+        \\  backslash
+        \a  alert (BEL)
+        \b  backspace (BS)
+        \c  produce no further output
+        \e  escape (ESC)
+        \f  form feed (FF)
+        \n  new line
+        \r  carriage return
+        \t  horizontal tab (HT)
+        \v  vertical tab (VT)
+"#; /* @MANEND */
+
+pub fn echo(args: &[String]) -> Result<(), io::Error> {
+    let mut arguments: Vec<&str> = vec![];
+    let mut data: Vec<&str> = vec![];
+
+    for arg in args {
+        if arg == "--help" {
+            arguments.push("help");
+        } else if arg.starts_with("-") {
+            let arg = &arg[1..];
+            for argopt in arg.chars() {
+                match argopt {
+                    'e' => arguments.push("escape"),
+                    'n' => arguments.push("no-newline"),
+                    's' => arguments.push("no-spaces"),
+                    'h' => arguments.push("help"),
```

Comment:
Could just have a single u8 represent all possible options, like so:

```rust
const ESCAPE: u8 = 1;
const NO_SPACE: u8 = 2;
const NO_NEWLINE: u8 = 4;

let mut flags = 0u8;
flags |= ESCAPE; // set the escape flag
if flags & (255 ^ ESCAPE) != 0 {
    // Executes if the escape flag was set
}
```

This also doesn't seem like it would function with long form options which are denoted with two -'s instead of one.